### PR TITLE
zotero: introduced support for endnote and footnote citation

### PR DESCRIPTION
**Pull request by lpranam**
Signed-off-by: Pranam Lashkari <lpranam@collabora.com>
Change-Id: I8429e81c58e72e846ce39de47e883ab6c96c8ebe

* Target version: master 

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/src/control/Control.Zotero.js [hunk 1]
```diff
@@ -319,6 +320,32 @@ L.Control.Zotero = L.Control.extend({
 		});
 	},
 
+	enableDialogFieldTypeCombobox: function(citationFormat) {
+		var entries = [];
+		if (citationFormat === 'note') {
```
Comment (timar) [inline browser/src/control/Control.Zotero.js:325]: interesting reuse of UI space for sematically different settings

**Comment by rriemann** [comment]
Very exciting features coming in here! :clap:

File: browser/src/control/Control.Zotero.js [hunk 2]
```diff
@@ -319,6 +320,32 @@ L.Control.Zotero = L.Control.extend({
...
+				jsontype: 'dialog',
+				action: 'update',
+				id: 'ZoteroDialog',
+				control: {
+					id: 'zoterotype',
+					type: 'combobox',
+					entries: entries,
+					selectedCount: '1',
+					selectedEntries: [
+						this.getFieldType() === 'Bookmark' ? 1 : 0
```
Comment (eszkadev) [inline browser/src/control/Control.Zotero.js:341]: you introduced more field types, it that still correct?

**Comment by eszkadev** [comment]
how I can test that? I don't see 'Footnotes' or 'Endnotes' in the UI. Do I need some special file?

File: browser/src/control/Control.Zotero.js (hunk 2 above)
Comment (lpranam) [reply browser/src/control/Control.Zotero.js:341]: yes still correct :)

**Comment by lpranam** [comment]
@eszkadev you have to select a particular style has citationFormat = note in its specification.

in style searchbox search for "(note)" to find such style

File: browser/src/control/Control.Zotero.js [hunk 3]
```diff
@@ -1045,19 +1045,21 @@ L.Control.Zotero = L.Control.extend({
 			return;
 		}
 		if (data.id == 'ok') {
-			// set selected type or field as default
+			// set selected style, style format and field
 			if (!this.selected || this.selected.type === 'style') {
 				var citationFormat = this.selected ? this.selected.citationFormat : this.settings.citationFormat;
-				this.setFieldType(this.selectedFieldType, citationFormat);
+				this.map.uiManager.showConfirmModal('zoterofieldtypekwarn', _('Citation warning'),
```
Comment (timar) [inline browser/src/control/Control.Zotero.js:1051]: zoterofieldtypekwarn
extra 'k' in id?

**Review by timar — CHANGES_REQUESTED** [review]
Did you disable the ability of changing the citation style in general?
I accept that you cannot change 'note' to something else (and vice versa?). But previously conversion between other types worked, right?

**Comment by lpranam** [comment]
@timar changing any kind of storage never worked (i.e: fields to bookmark or vice versa) and same goes for every combination.

**Review by timar — APPROVED** [review]

**Pull request merged**